src/atomate2/abinit/flows/core.py

```diff
@@ -0,0 +1,195 @@
+"""Core abinit flow makers."""
+
+import copy
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import List, Optional, Union
+
+from jobflow import Flow, Maker
+from pymatgen.core.structure import Structure
+
+from atomate2.abinit.jobs.base import BaseAbinitMaker
+from atomate2.abinit.jobs.core import NonSCFMaker, RelaxMaker, StaticMaker
+
+
+@dataclass
+class BandStructureMaker(Maker):
+    """
+    Maker to generate abinit band structures.
+
+    This is a static calculation followed by two non-self-consistent field
+    calculations, one uniform and one line mode.
+
+    Parameters
+    ----------
+    name : str
+        Name of the flows produced by this maker.
+    scf_maker : .BaseAbinitMaker
+        The maker to use for the static calculation.
+    bs_maker : .BaseAbinitMaker
+        The maker to use for the non-self-consistent field calculations.
+    """
+
+    name: str = "band structure"
+    bandstructure_type: str = "both"
+    static_maker: BaseAbinitMaker = field(default_factory=StaticMaker)
+    bs_maker: BaseAbinitMaker = field(default_factory=NonSCFMaker)
+
+    def make(
+        self,
+        structure: Structure,
+        restart_from: Optional[Union[str, Path]] = None,
+    ):
+        """
+        Create a band structure flow.
+
+        Parameters
+        ----------
+        structure : Structure
+            A pymatgen structure object.
+        restart_from : str or Path or None
+            One previous directory to restart from.
+
+        Returns
+        -------
+        Flow
+            A band structure flow.
+        """
+        if self.bandstructure_type not in ("both", "line", "uniform"):
+            raise ValueError(
+                f"Unrecognised bandstructure type {self.bandstructure_type}"
+            )
+
+        static_job = self.static_maker.make(structure, restart_from=restart_from)
+        jobs = [static_job]
+
+        if self.bandstructure_type in ("both", "uniform"):
+            uniform_maker = copy.deepcopy(self.bs_maker)
```

Comment:
I have to make some checks but I think that when using the run_locally, because the objects are the same, you basically run the same calculation twice instead of doing one uniform and one line (when bandstructure_type is "both").

Reply:
Works fine without the copy. Maybe there was another issue that was fixed since.